src/features/contact/contactverzoek/overzicht/ContactverzoekenZoeker.vue

```diff
@@ -13,100 +13,119 @@
-import { ensureState } from "@/stores/create-store"; //todo: niet in de stores map. die is applicatie specifiek. dit is generieke functionaliteit
-import { useRouter } from "vue-router";
+import { computed, ref, onMounted } from "vue";
+import SimpleSpinner from "@/components/SimpleSpinner.vue";
 import SearchResultsCaption from "@/components/SearchResultsCaption.vue";
 import { Button as UtrechtButton } from "@utrecht/component-library-vue";
 import ContactverzoekenOverzicht from "./ContactverzoekenOverzicht.vue";
+import { ensureState } from "@/stores/create-store";
+import ContactmomentDetailsContext from "@/features/contact/contactmoment/ContactmomentDetailsContext.vue";
+import ContactmomentPreview from "@/features/contact/contactmoment/ContactmomentPreview.vue";
+import { useOpenKlant2 } from "@/services/openklant2";
+import { search } from "./service";
+import type { Contactverzoek } from "./types";
+import type { PaginatedResult } from "@/services";
+import ZaakPreview from "@/features/zaaksysteem/components/ZaakPreview.vue";
+import ApplicationMessage from "@/components/ApplicationMessage.vue";
+
+const openKlant2 = ref<boolean>(false);
 
 const store = ensureState({
-  stateId: "klant-zoeker",
+  stateId: "contactverzoeken-zoeker",
   stateFactory() {
     return {
-      // currentPhone: "",
-      // currentEmail: "",
       searchQuery: "",
-      // klantPhone: "",
-      // klantEmail: "",
+      zoekerResults: {
+        loading: false,
+        success: false,
+        error: false,
+        data: [] as PaginatedResult<Contactverzoek>[],
+      },
     };
   },
 });
 
-const query = ref<string>("");
-
-const q = computed(() => ({ query: store.value.searchQuery }));
+onMounted(async () => {
+  openKlant2.value = await useOpenKlant2();
+});
 
-const zoeker = useSearch(q);
+const handleSearch = async () => {
+  store.value.zoekerResults.loading = true;
+  store.value.zoekerResults.success = false;
+  store.value.zoekerResults.error = false;
 
-const singleKlantId = computed(() => {
-  if (zoeker.success && zoeker.data.length === 1) {
-    const first = zoeker.data[0];
-    if (first?._typeOfKlant === "klant") {
-      return first.id;
-    }
+  try {
+    store.value.zoekerResults.data = await search(
+      store.value.searchQuery,
+      openKlant2,
+    );
+    store.value.zoekerResults.success = true;
+  } catch (error) {
+    store.value.zoekerResults.error = true;
+  } finally {
+    store.value.zoekerResults.loading = false;
   }
-  return undefined;
-});
+};
 
 const filteredZoekerData = computed(() => {
-  if (zoeker.success) {
-    return zoeker.data.filter(
-      (item) =>
-        !item.record.data.betrokkene.hasOwnProperty.call(
-          item.record.data.betrokkene,
-          "klant",
+  if (store.value.zoekerResults.success && store.value.searchQuery) {
+    // Bij OK2 (klantinteracties) flow checken we op wasPartij
+    if (openKlant2.value) {
```

Comment:
Misschien is het netter om het uitfilteren al in de service op te lossen? Dan heb je hier geen computed meer nodig en hoef je niet op meerdere plekken te checken of je in ok1 of 2 zit

Reply:
aangepast